### PR TITLE
Build fixes

### PR DESCRIPTION
**Pull request by neheb**
https://downloads.openwrt.org/snapshots/faillogs/arc_archs/routing/

### CODE REVIEW

**Comment by ecsv** [comment]
@jow- @mwarning @lgierth @dangowrt 
@stepet92 @Noltari @sbyx

File: mcproxy/patches/0003-add-sourcefilter.patch [hunk 1]
```diff
@@ -4,7 +4,7 @@
  #include <numeric>
  #include <unistd.h>
  
-+#ifndef __GLIBC__
++#if !defined(__GLIBC__) || defined(__UCLIBC__)
```
Comment (jow-) [inline mcproxy/patches/0003-add-sourcefilter.patch]: Is this really supposed to mean "not glibc or uclibc"? Or did you rather mean `#if  !defined(__GLIBC__) && !defined(__UCLIBC__)` ?

Comment (cotequeiroz) [reply mcproxy/patches/0003-add-sourcefilter.patch]: I haven't tested this, but reading the commit message it looks correct.  
uClibc-ng defines `__GLIBC__` but needs to include `sourcefilter.cpp`.

Comment (neheb) [reply mcproxy/patches/0003-add-sourcefilter.patch]: It's correct. It's needed for uclibc-ng but not glibc.

**Comment by neheb** [comment]
Any other problems?

**Pull request merged**